### ISSUE TITLE
Nebula Lint does not handle certain `@Retention(RetentionPolicy.SOURCE)` annotations correctly

### ISSUE THREAD

**Issue by jsfzoloz** [nebula-plugins/gradle-lint-plugin#170]
#### Summary:
Nebula Lint should not warn of an unused dependency for `@Retention(RetentionPolicy.SOURCE)` annotations . 

#### Steps to reproduce:
Create a gradle project and add the gradle lint plugin. Add project lombok. Create a class and annotate it with a lombok annotation. Run gradle lint and observe the warning.

#### Extended Description:
Project Lombok is an annotation processor that will generate getters/setters for your code at compile time. Take a look at their `@Data` annotation source:

```
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.SOURCE)
public @interface Data {
}
```

Here is an example class using that
```
@Data
public class SomeDataStruct {
  private Integer id;
  private String name;
}
```

Nebula lint gives the following warning:
```
> Task :lintGradle FAILED

This project contains lint violations. A complete listing of the violations follows. 
Because none were serious, the build's overall status was unaffected.

warning   unused-dependency                  this dependency is a service provider unused at compile time and can be moved to the runtime configuration
```

#### Workaround:
```
	gradleLint.ignore('unused-dependency') {
		compileOnly group: 'org.projectlombok', name: 'lombok', version:'1.16.20'
	}
```

**Comment by dbolger** [comment]
See my comment about this here: https://github.com/nebula-plugins/gradle-lint-plugin/issues/376#issuecomment-2096010123
